### PR TITLE
chore!: ci-2162 remove node<18 support and fix tests

### PR DESCRIPTION
**Pull request by debugwand**
* remove node 14 and 16 and replace with 18 and 20
* tests were failing, and using enzyme which doesn't work beyond node 16, so I've replaced this with react testing library. I've also adjusted the tests as they were just testing that the data was being added to the props in a shallow rendering of the component which isn't all that useful. Instead, I've made sure specific bits of the config get output to the HTML.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -31,14 +31,14 @@
 		"@financial-times/o-colors": "^6.4.0",
 		"@financial-times/o-grid": "^6.1.1",
 		"@financial-times/o-spacing": "^3.0.0",
-		"@financial-times/o-typography": "^7.2.0"
+		"@financial-times/o-typography": "^7.2.0",
+		"@dotcom-tool-kit/circleci": "^6.0.1"
```
Comment (umbobabo) [inline package.json:35]: Nitpick: Shouldn't this be a dev dep?

File: src/Eventpromo.jsx [hunk 2]
```diff
@@ -21,7 +21,7 @@ const Eventpromo = ({...props}) => {
 const Forumpromo = ({...props}) => {
 	const branding = getBranding('ft-forums')
 	return (
-		<div className={styles.eventpromo} data-forum-id={props.id}>
+		<div className={styles.eventpromo} data-forum-id={props.id} data-testid="forumpromo">
```
Comment (umbobabo) [inline src/Eventpromo.jsx:24]: Thought: Assuming this data attribute will stay in Production except if there is some transpiring happening, I don't personally like the approach. It adds mark-up attributes just for sake of testing. I would instead target existing data attributes if possible. However, do not spend to much time fixing this 👍 , it doesn't worth it.

File: tooling/babel/index.js [hunk 3]
```diff
@@ -26,7 +24,7 @@ exports.server = {
 			'@babel/preset-env',
 			{
 				targets: {
-					node: 12
+					node: 16
```
Comment (umbobabo) [inline tooling/babel/index.js:27]: Question: shouldn't be this node 18?

File: src/Eventpromo.jsx (hunk 2 above)
Comment (debugwand) [reply src/Eventpromo.jsx:24]: I agree, I got stuck on the `screen.getBy...` options being too limited - is there a better way we prefer?

Comment (umbobabo) [reply src/Eventpromo.jsx:24]: Leave it like this for now, no problem

**Review by umbobabo — APPROVED** [review]

**Pull request merged**